<?xml version="1.0" ?>

<launch>

<!-- script bash -->
	<node pkg="rccar" type="jetson.sh" name="jetson_script" output="screen" />

<!--Robot Model-->
	  <!-- group for racecar -->
	<group ns="racecar">
	  <!--<param name="tf_prefix" value="racecar"/>-->
	<param name="robot_description" command="xacro --inorder '$(find rccar)/description/racecar_fixed_wheel.xacro'"/>
	  <!-- Add the transformations -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
	</group>

<!-- LIDAR -->
	<node name="urg_node" pkg="urg_node" type="urg_node">
		    <param name="ip_address" value="10.42.0.18"/>
		    <param name="range_min" value="0.25"/>
	</node>

<!-- VESC driver node -->

	<rosparam file="$(find rccar)/config/vesc.yaml" command="load" />
	  <!-- Optionally launch in GDB, for debugging -->
	<arg name="debug" default="false" />
	<arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
	<arg unless="$(arg debug)" name="launch_prefix" value="" />
	  
	<node pkg="vesc_driver" type="vesc_driver_node" name="vesc_driver_node"
		output="screen" launch-prefix="$(arg launch_prefix)" >
	    	<param name="port" value="/dev/ttyACM0" />
	</node>

<!-- Ackermann to VESC node -->
	<node pkg="vesc_ackermann" type="ackermann_to_vesc_node" name="ackermann_to_vesc_node"
		output="screen" launch-prefix="$(arg launch_prefix)" >
	</node>
  
<!-- RVIZ plugins -->
	<node pkg="rccar" name="rviz_overlay_node" type="rviz_overlay.py" output="screen">
	</node>

<!-- VESC to odom node -->
	<node pkg="rccar" name="vesc_to_odom_node" type="vesc_to_odom.py" output="screen"> 
		<!--<remap from="/odom" to="/vesc/odom"/>-->
	</node>


<!-- USB CAM -->
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
	    <param name="video_device" value="/dev/video0" />
	    <param name="image_width" value="640" />
	    <param name="image_height" value="480" />
	    <param name="pixel_format" value="yuyv" />
	    <param name="color_format" value="yuv422p" />
	    <param name="camera_frame_id" value="usb_cam" />
	    <param name="io_method" value="mmap"/>
	</node>
	
<!--RObot localization-->
	<!--<node pkg="robot_localization" type="ekf_localization_node" name="ekf_localisation_node" clear_params="true">
    	<rosparam command="load" file="$(find rccar)/config/ekf.yaml" />
	      Placeholder for output topic remapping
	    <remap from="odometry/filtered" to="ekf/odom"/>
	    <remap from="accel/filtered" to="ekf/accel"/>
	    
	</node> -->
	
<!-- GMAPPING SLAM -->
	<!--<node pkg="gmapping" name="slam_gmapping" type="slam_gmapping">
		<param name="scan" value="scan" />
		<param name="odom_frame" value="vesc_odom" />
		<param name="map_update_interval" value="2.0" />



		    <param name="maxUrange" value="6.0"/> 
		    <param name="maxRange" value="8.0"/> 
		    <param name="sigma" value="0.05"/> 
		    <param name="kernelSize" value="1"/> 
		    <param name="lstep" value="0.05"/> 
		    <param name="astep" value="0.05"/> 
		    <param name="iterations" value="5"/> 
		    <param name="lsigma" value="0.075"/> 
		    <param name="ogain" value="3.0"/> 
		    <param name="lskip" value="0"/> 
		    <param name="minimumScore" value="100"/> 
		    <param name="srr" value="0.01"/> 
		    <param name="srt" value="0.02"/> 
		    <param name="str" value="0.01"/> 
		    <param name="stt" value="0.02"/> 
		    <param name="linearUpdate" value="0.5"/> 
		    <param name="angularUpdate" value="0.436"/> 
		    <param name="temporalUpdate" value="-1.0"/> 
		    <param name="resampleThreshold" value="0.5"/> 
		    <param name="particles" value="80"/> 
		  
		    <param name="xmin" value="-1.0"/> 
		    <param name="ymin" value="-1.0"/> 
		    <param name="xmax" value="1.0"/> 
		    <param name="ymax" value="1.0"/> 
		 
		    <param name="delta" value="0.05"/> 
		    <param name="llsamplerange" value="0.01"/> 
		    <param name="llsamplestep" value="0.01"/> 
		    <param name="lasamplerange" value="0.005"/> 
		    <param name="lasamplestep" value="0.005"/> 

		
	</node> -->

</launch>
